tools/toolchains/arm.py

```diff
@@ -56,6 +56,11 @@ def __init__(self, target, options=None, notify=None, macros=None, silent=False,
         else:
             cpu = target.core
 
+        if not TOOLCHAIN_PATHS['ARM']:
+            exe = find_executable('armcc')
+            if exe:
```

Comment:
@bogdanm @theotherjimmy has strong opinions here. I agree that we are delaying failure. But I think it is generally assumed that when you are constructing an object, it will succeed. Though I think __init__ is not a constructor in the same way as C++.